python/taichi/lang/impl.py

```diff
@@ -213,8 +213,11 @@ def layout():
         taichi_lang_core.layout(layout)
         self.materialized = True
         for var in self.global_vars:
-            assert var.ptr.snode(
-            ) is not None, 'Some variable(s) are not placed'
+            if var.ptr.snode() is None:
+                raise RuntimeError(
+                    'Some variable(s) are not placed.'
+                    ' Did you forget to specify shape for a field? e.g.:\n'
```

Comment:
If `shape` is a required parameter now, then it's not possible for users to forget to specify `shape`. Instead I think this case would occur when users did specify `shape`, but forgot to do a `place()`.